src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,181 @@
+      false);
+
+  public ShooterSubsystem() {
+    var shooterMotorConfig = new TalonFXConfiguration();
+    shooterMotorConfig.Slot0 = Slot0Configs.from(SHOOTER_SLOT_CONFIGS);
+    shooterMotorConfig.MotorOutput.NeutralMode = Coast;
+    shooterMotorConfig.OpenLoopRamps.DutyCycleOpenLoopRampPeriod = 1;
+    shooterMotorConfig.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+    shooterLeftMotor.getConfigurator().apply(shooterMotorConfig);
+    shooterMotorConfig.MotorOutput.Inverted = InvertedValue.CounterClockwise_Positive;
+    shooterRightMotor.getConfigurator().apply(shooterMotorConfig);
+
+    // Get the through-bore-encoder absolute encoder
+    actuatorEncoder = actuatorMotor.getAbsoluteEncoder(kDutyCycle);
+    actuatorEncoder.setInverted(true);
+    actuatorEncoder.setAverageDepth(64);
+    pidController = actuatorMotor.getPIDController();
+    pidController.setFeedbackDevice(actuatorEncoder);
+
+  }
+
+  private static SparkPIDController indexerMotorConfig(CANSparkMax sparkMax, boolean invert) {
+    SparkPIDController pidController = sparkMax.getPIDController();
+    sparkMax.restoreFactoryDefaults();
+    sparkMax.enableVoltageCompensation(12);
+    sparkMax.setOpenLoopRampRate(0.1);
+    sparkMax.setClosedLoopRampRate(0.1);
+    sparkMax.setInverted(invert);
+    pidController.setP(IndexerConstants.BELT_kP);
+    pidController.setFF(0.00009);
+    sparkMax.getEncoder();
+    sparkMax.burnFlash();
+    sparkMax.setIdleMode(IdleMode.kCoast);
+    return pidController;
+
+  }
+
+  public void spinShooterWheel(double shooterRPS) {
+    shooterLeftMotor.setControl(shooterMotorVelocity.withVelocity(shooterRPS));
+    shooterRightMotor.setControl(shooterMotorVelocity.withVelocity(shooterRPS));
+  }
+
+  public Command sysIdShooterMotorQuasiCommand(Direction direction) {
+    return shooterMotorSysIdRoutine.quasistatic(direction).withName("SysId Shooter Motors Quasistatic " + direction)
+        .finallyDo(this::stop);
+  }
+
+  public Command sysIdShooterMotorDynamCommand(Direction direction) {
+    return shooterMotorSysIdRoutine.dynamic(direction).withName("SysId Shooter Motors Quasistatic " + direction)
+        .finallyDo(this::stop);
+  }
+
+  public void activeStop() {
+    if (!isActiveStopped) {
+      shooterLeftMotor.setControl(shooterMotorVelocity.withVelocity(0));
+      shooterRightMotor.setControl(shooterMotorVelocity.withVelocity(0));
+    }
+    isActiveStopped = true;
+  }
+
+  /**
+   * Gets the wrist position Zero is horizontal, up is positive
+   * 
+   * @return position in radians
+   */
+  public double getWristPosition() {
+    return Units.rotationsToRadians(actuatorEncoder.getPosition() + ENCODER_OFFSET);
+  }
+
+  public void actuatorUp(Measure<Voltage> volts) {
+    indexerMotorConfig(actuatorMotor, true);
+    actuatorMotor.setVoltage(volts.in(Volts));
+  }
+
+  public void actuatorDown(Measure<Voltage> volts) {
+    indexerMotorConfig(actuatorMotor, false);

```

Comment:
Controllers should not be configured on the fly. Just apply negative output to go down.